### PR TITLE
Make it build with ghc 9.8

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: jsapi/test/Main.hs [hunk 1]
```diff
@@ -31,8 +31,8 @@ foreign import javascript interruptible "testStart($1, $c);" initComplete
 #else
 initComplete :: Object -> JSM ()
 initComplete api =
-  void $ api ^. js1 "version" (fun $ \ _ _ [ver] ->
-    fromJSValUnchecked ver >>= (liftIO . putStrLn))
+  void $ api ^. js1 "version" (fun $ \ _ _ xs ->
+    fromJSValUnchecked (head xs) >>= (liftIO . putStrLn))
```
[the remark this answers is not in the document]
Comment (erikd) [reply jsapi/test/Main.hs:35]: This is a change I made in August. Will review.

File: cabal.project [hunk 2]
```diff
@@ -56,10 +60,6 @@ else
 
   active-repositories: hackage.haskell.org, cardano-haskell-packages
 
-index-state:
-  , hackage.haskell.org 2024-06-05T16:48:53Z
-  , cardano-haskell-packages 2024-06-05T16:48:53Z
-
```
[the remark this answers is not in the document]
Comment (erikd) [reply cabal.project:62]: Oops, that was probably not intended. A git rebase/merge stuff up.

File: jsapi/test/Main.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (angerman) [reply jsapi/test/Main.hs:35]: 👍

**Comment by erikd** [comment]
@paweljakubas Please review this and merge if you are happy with it.

**Pull request merged**